### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by JakeLin**
Fix some typo, improve code quality and etc.

### CODE REVIEW

File: YYKit/Base/Foundation/NSArray+YYAdd.h [hunk 1]
```diff
@@ -35,7 +35,7 @@
 /**
  Serialize the array to a binary property list data.
  
- @return A bplist data, or nil if an error occurs.
+ @return A plist data, or nil if an error occurs.
```
Comment (ibireme) [inline YYKit/Base/Foundation/NSArray+YYAdd.h]: bplist means binary plist

File: YYKit/Base/Foundation/NSNumber+YYAdd.m [hunk 2]
```diff
@@ -35,9 +35,9 @@ + (NSNumber *)numberWithString:(NSString *)string {
                 @"null" :   [NSNull null],
                 @"<null>" : [NSNull null]};
     });
-    NSNumber *num = dic[str];
+    id num = dic[str];
     if (num) {
-        if (num == (id)[NSNull null]) return nil;
+        if ([num isMemberOfClass:[NSNull class]]) return nil;
```
Comment (ibireme) [inline YYKit/Base/Foundation/NSNumber+YYAdd.m]: NSNull is a singleton, so it can be compared directly

**Comment by ibireme** [comment]
Thank you for your commits!

File: YYKit/Base/Foundation/NSArray+YYAdd.h (hunk 1 above)
Comment (JakeLin) [inline YYKit/Base/Foundation/NSArray+YYAdd.h]: Thanks, didn't get that. can you change it to "binary plist"

File: YYKit/Base/Foundation/NSNumber+YYAdd.m (hunk 2 above)
Comment (JakeLin) [inline YYKit/Base/Foundation/NSNumber+YYAdd.m]: Thanks, get it. "singleton object used to represent null" from http://nshipster.com/nil/

**Comment by JakeLin** [comment]
Thanks, created another PR https://github.com/ibireme/YYKit/pull/16

**Pull request merged**